RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc

```diff
@@ -210,6 +213,17 @@ void PhotonProducer::produce(edm::Event& theEvent, const edm::EventSetup& theEve
   }
   if( validEcalRecHits) endcapRecHits = *(endcapHitHandle.product());
 
+  bool validPreshowerRecHits=true;
+  Handle<EcalRecHitCollection> preshowerHitHandle;
+  theEvent.getByToken(preshowerHits_, preshowerHitHandle);
+  EcalRecHitCollection preshowerRecHits;
+  if (!preshowerHitHandle.isValid()) {
+    edm::LogError("PhotonProducer")
+      << "Error! Can't get the preshowerEcalHits";
+    validPreshowerRecHits=false;
+  }
```

Comment:
This was the request made by @Sam-Harper .  I guess we would need a config anyways to protect against the toolsForES at L454.  Thoughts?